src/uwtools/rocoto.py

```diff
@@ -183,12 +183,26 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
         :param config: Configuration data for this element.
         """
         e = SubElement(e, STR.dependency)
+        for key in config.keys():
+            tag, _ = self._tag_name(key)
+            if tag in [STR.taskdep, STR.datadep, STR.timedep]:
+                self._add_task_dependency_operand(e, config=config)
+            else:
+                raise UWConfigError("Unhandled dependency type %s" % tag)
+
+    def _add_task_dependency_operand(self, e: Element, config: dict) -> None:
+        """
+        ???
+        """
         for key, block in config.items():
             tag, _ = self._tag_name(key)
             if tag == STR.taskdep:
                 self._set_attrs(SubElement(e, STR.taskdep), block)
+            elif tag == STR.datadep:
+                self._set_attrs(SubElement(e, STR.datadep), config={"attrs": block})
             else:
-                raise UWConfigError("Unhandled dependency type %s" % tag)
+                pass
+                # self._set_attrs(SubElement(e, STR.timedep), config={"attrs": block})
```

Comment:
Something else I am currently working on is the `timedep` tag. Here is an example of what a `timedep` tag would look like 
`<timedep><cyclestr offset="&DEADLINE;">@Y@m@d@H@M@S</cyclestr></timedep>`

We will not be handling `cyclestr` until a later ticket planned for sprint 10.5. I think for now the idea would be to just include 
`<timedep> offset="&DEADLINE;"</timedep>`
as an example of what should be covered by this ticket.

Reply:
Seems reasonable for now, as long as it passes the RELAX NG schema check: Is it ok with `<timedep>` having arbitrary text content instead of a `<cyclestr>` child?

Reply:
It is currently happy with the arbitrary text instead of a ```<cyclestr>``` child and passes the RELAXNG schema check.

We will need to add this note to add additional criteria to the current Rocoto XML ticket that covers ```<cyclestr>``` or create a separate ticket since one involves modifying the JSONSchema and this requires updating the ```<dependency>``` element here. I feel like there is also an order to first modifying the JSONSchema and then completing this task. This should be a very minor change once the schema is updated.

Reply:
I agree with you about updating the schema first, then implementing the code generation. Considering our sprint review yesterday, separate small tasks is probably preferable to one larger one.